rich/pretty.py

```diff
@@ -79,6 +79,24 @@ def _is_dataclass_repr(obj: object) -> bool:
         return False
 
 
+_dummy_namedtuple = collections.namedtuple("_dummy_namedtuple", [])
+
+
+def _has_default_namedtuple_repr(obj: object) -> bool:
+    """Check if an instance of namedtuple contains the default repr
+
+    Args:
+        obj (object): A namedtuple
+
+    Returns:
+        bool: True if the default repr is used, False if there's a custom repr.
+    """
+    obj_file = inspect.getfile(obj.__repr__)
```

Comment:
This can throw a `TypeError` according to the docs. Is that an issue here?

Reply:
I've handled `TypeError` and `OSError` now.